datafusion/physical-expr/src/datetime_expressions.rs

```diff
@@ -1337,6 +1376,36 @@ fn validate_to_timestamp_data_types(
     None
 }
 
+/// to_date SQL function implementation
+pub fn to_date_invoke(args: &[ColumnarValue]) -> Result<ColumnarValue> {
+    if args.is_empty() {
+        return exec_err!(
+            "to_date function requires 1 or more arguments, got {}",
+            args.len()
+        );
+    }
+
+    // validate that any args after the first one are Utf8
+    if args.len() > 1 {
+        if let Some(value) = validate_to_timestamp_data_types(args, "to_date") {
+            return value;
+        }
+    }
+
+    match args[0].data_type() {
+        DataType::Int32
```

Comment:
It looks like your code takes several input types -- can you please add tests that show calling `to_date` with these types (eg `select to_date(arrow_cast(123, 'Int64'))`)

Reply:
Done